sms_wsj/database/create_intermediate_json.py

```diff
@@ -183,21 +183,29 @@ def extend_composition_example_greedy(rng, speaker_ids, example_compositions=Non
 
 
 def get_randomized_example(
-    rir_example, source_examples, rng, dataset_name, database_path
+    rir_example, source_examples, rng, dataset_name
 ):
+    example_id = "_".join([
+        rir_example['example_id'],
+        *[source_ex["example_id"] for source_ex in source_examples],
+    ])
+    rng = get_rng(dataset_name, example_id)
+
     example = copy(rir_example)
+    example['example_id'] = example_id
+    example['dataset'] = dataset_name
+
     assert len(source_examples) <= len(example['source_position'][0])
     num_speakers = len(source_examples)
 
+    # Remove unused source positions and rirs (i.e. the scenarios.json was
+    # maybe generated with more speakers)
     example['source_position'] = [
-        source_position[:num_speakers]
-        for source_position in example['source_position']
-    ]
+        v[:num_speakers] for v in example['source_position']]
```

Comment:
Why is the speaker dimension the second dimension? Shouldnt it be the first?

Reply:
I can double-check, but I think it was the last. Maybe the ordering was motivated by wsj0-2mix or our old MATLAB code.